templates/skeleton/app/components/CartMain.tsx

```diff
@@ -26,7 +26,7 @@ export function CartMain({layout, cart: originalCart}: CartMainProps) {
     cart &&
     Boolean(cart?.discountCodes?.filter((code) => code.applicable)?.length);
   const className = `cart-main ${withDiscount ? 'with-discount' : ''}`;
-  const cartHasItems = !!cart && cart.totalQuantity > 0;
+  const cartHasItems = cart?.totalQuantity && cart.totalQuantity > 0;
```

Comment:
Maybe rely on the gt operator to narrow the type to `boolean` instead of `boolean | number | undefined`?

```suggestion
  const cartHasItems = cart?.totalQuantity > 0;
```

Reply:
Typescript doesn't like `undefined > 0`. So it has to be `cart?.totalQuantity! > 0`